lib/python/qmk/errors.py

```diff
@@ -1,6 +1,17 @@
-class NoSuchKeyboardError(Exception):
-    """Raised when we can't find a keyboard/keymap directory.
+class DirectoryError(Exception):
```

Comment:
I added a generically named error to raise directory type errors. I used this DirectoryError exception to raise an error to tell the user a path did/didn't exist.

May make more sense to change this to something more generic like `class QmkError(Exception):` and create a new error class that inherits it, like `class DirectoryError(QmkError):`